src/state/reducers/workspace/WorkspaceReducer.js

```diff
@@ -33,10 +33,18 @@ const workspacesListInitialState = {
 };
 
 const workspacesListReducer = createReducer(workspacesListInitialState, (builder) => {
-  builder.addCase(WORKSPACE_ACTIONS_KEY.SET_ALL_WORKSPACES, (state, action) => {
-    state.data = action.list;
-    state.status = action.status;
-  });
+  builder
+    .addCase(WORKSPACE_ACTIONS_KEY.LINK_TO_DATASET, (state, action) => {
+      if (action.datasetId == null) return;
+
+      const targetWorkspace = state.data.find((workspace) => workspace.id === action.workspaceId);
+      const linkedDatasets = targetWorkspace?.linkedDatasetIdList;
+      if (linkedDatasets !== undefined) linkedDatasets.push(action.datasetId);
```

Comment:
just in case: `linkedDatasets` cannot be `null` ?

Reply:
You're right, I think it can be `null` for "legacy" workspaces, I'll replace the condition by `linkedDatasets != null` to handle it correctly. Thank you ! :smile: